dev/NavigationView/NavigationView.cpp

```diff
@@ -1889,15 +1966,23 @@ void NavigationView::RepeaterGettingFocus(const winrt::IInspectable& sender, con
             // If focus is coming from outside the root repeater, put focus on last focused item
             if (rootRepeater != oldElementParent)
             {
-                if (auto const argsAsIGettingFocusEventArgs2 = args.try_as<winrt::IGettingFocusEventArgs2>())
+                if (auto const lastFocusedNvi = rootRepeater.TryGetElement(m_indexOfLastFocusedItem))
                 {
-                    if (auto const lastFocusedNvi = rootRepeater.TryGetElement(m_indexOfLastFocusedItem))
+                    if (auto const argsAsIGettingFocusEventArgs2 = args.try_as<winrt::IGettingFocusEventArgs2>())
                     {
                         if (argsAsIGettingFocusEventArgs2.TrySetNewFocusedElement(lastFocusedNvi))
                         {
                             args.Handled(true);
                         }
                     }
+                    else
+                    {
+                        // Without TrySetNewFocusedElement, we cannot set focus while it is changing.
+                        m_dispatcherHelper.RunAsync([lastFocusedNvi]()
+                            {
+                                SetFocus(lastFocusedNvi, winrt::FocusState::Keyboard);
```

Comment:
>Keyboard [](start = 76, length = 8)

probably programmatic is better

Reply:
see https://github.com/microsoft/microsoft-ui-xaml/pull/1850#discussion_r368166014